packages/intrepid2/src/Projection/Intrepid2_ProjectionTools.hpp

```diff
@@ -780,15 +780,27 @@ class ProjectionTools {
       Teuchos::LAPACK<ordinal_type,valueType> lapack_;
       ordinal_type numCells = basisCoeffs.extent(0);
 
+      // due to the particular type of elemDof (possibly due either to layout or subviews having been
+      // taken), the usual pattern of create_mirror_view_and_copy() does not work "out of the box".
+      // So we do a manual copy on device, followed by the create_mirror_view_and_copy().
+      Kokkos::View<ordinal_type*> elemDofCopy("elemDofCopy", elemDof.extent(0));
+      Kokkos::parallel_for ("Intrepid2::ProjectionTools - copy elemDof",
+                            Kokkos::RangePolicy<ExecSpaceType, int> (0, elemDof.extent(0)),
+      KOKKOS_LAMBDA (const unsigned &i) {
+        elemDofCopy(i) = elemDof(i);
+      });
+      auto serialElemDof = Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace(), elemDofCopy);
+      
+      Kokkos::DynRankView<typename ViewType2::value_type, HostDeviceType> A_host("A0_host", elemMat.extent(1),elemMat.extent(2));
+      auto A_device = Kokkos::create_mirror_view(typename DeviceType::memory_space(), A_host);
+
       if(matrixIndependentOfCell_) {
         ViewType2 elemRhsTrans("transRhs", elemRhs.extent(1), elemRhs.extent(0));
         Kokkos::View<valueType**,Kokkos::LayoutLeft,HostDeviceType>
         pivVec("pivVec", m+n + std::max(m+n, numCells), 1);
 
         Kokkos::View<valueType**,Kokkos::LayoutLeft,HostDeviceType> serialElemRhs("serialElemRhs", n+m, numCells);
 
-        Kokkos::DynRankView<typename ViewType2::value_type, HostDeviceType> A_host("A0_host", elemMat.extent(1),elemMat.extent(2));
-        auto A_device = Kokkos::create_mirror_view(typename DeviceType::memory_space(), A_host);
```

Comment:
When the layout does not match here, use the derived layout from device type. We don't need to run the manual copy.

Reply:
Also when we create the A_host, it can be created appropriately as a mirror view having the same memory layout of elemMat

Reply:
@kyungjoo-kim, I'm not sure I follow.  Could you give a code snippet?